controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1355,6 +1357,87 @@ func (r *BareMetalHostReconciler) actionDeprovisioning(prov provisioner.Provisio
 	return actionComplete{}
 }
 
+func (r *BareMetalHostReconciler) checkServicing(prov provisioner.Provisioner, info *reconcileInfo, hup *metal3api.HostUpdatePolicy) (result actionResult, isServicing bool) {
+	servicingData := provisioner.ServicingData{}
+
+	var fwDirty bool
+	var hfsDirty bool
+
+	// (NOTE)janders: since Servicing is an opt-in feature that requires HostUpdatePolicy to be created and set to onReboot
+	// set below booleans to false by default and change to true based on policy settings
+
+	servicingData.LiveFirmwareUpdateAllowed = false
+	servicingData.LiveFirmwareSettingsAllowed = false
+
+	if hup != nil {
+		if hup.Spec.FirmwareSettings == metal3api.HostUpdatePolicyOnReboot {
+			servicingData.LiveFirmwareSettingsAllowed = true
+		}
+		if hup.Spec.FirmwareUpdates == metal3api.HostUpdatePolicyOnReboot {
+			servicingData.LiveFirmwareUpdateAllowed = true
+		}
+	}
+
+	if servicingData.LiveFirmwareUpdateAllowed {
```

Comment:
Btw, the way code is written, you don't need to store the *Allowed variables in servicingData: you fully use them locally.